torchtext/experimental/datasets/sequence_tagging.py

```diff
@@ -0,0 +1,220 @@
+                              totensor(dtype=torch.long))
+        for idx in range(len(vocabs) - 1)
+    ]
+    transformers = [word_transform, *labels_transforms]
+
+    datasets = []
+    for item in data_select:
+        if raw_data[item] is not None:
+            datasets.append(
+                SequenceTaggingDataset(raw_data[item], vocabs, transformers))
+
+    return datasets
+
+
+class SequenceTaggingDataset(torch.utils.data.Dataset):
+    """Defines an abstraction for raw text sequence tagging iterable datasets.
+    Currently, we only support the following datasets:
+        - UDPOS
+        - CoNLL2000Chunking
+    """
+    def __init__(self, data, vocab, transforms):
+        """Initiate sequence tagging dataset.
+        Arguments:
+            data: a list of word and its respective tags. Example:
+                [[word, POS, dep_parsing label, ...]]
+            vocab: Vocabulary object used for dataset.
+            transforms: a list of string transforms for words and tags.
+        """
+
+        super(SequenceTaggingDataset, self).__init__()
+        self.data = data
+        self.vocab = vocab
+        self.transforms = transforms
+
+    def __getitem__(self, i):
+        line = []
+        for idx, transform in enumerate(self.transforms):
+            line.append(transform(self.data[i][idx]))
+        return line
+
+    def __len__(self):
+        return len(self.data)
+
+    def get_vocab(self):
+        return self.vocab
+
+
+def UDPOS(train_filename="en-ud-tag.v2.train.txt",
+          valid_filename="en-ud-tag.v2.dev.txt",
+          test_filename="en-ud-tag.v2.test.txt",
+          data_select=("train", "valid", "test"),
+          root=".data",
+          vocabs=None,
+          word_tokenizer=None):
+    """ Universal Dependencies English Web Treebank
+
+    Separately returns the training and test dataset
+
+    Arguments:
+        train_filename: Filename for training dataset.
+            Default: en-ud-tag.v2.train.txt
+        valid_filename: Filename for validation dataset.
+            Default: en-ud-tag.v2.dev.txt
+        test_filename: Filename for test dataset.
+            Default: en-ud-tag.v2.test.txt
+        data_select: a string or tuple for the returned datasets
+            (Default: ('train', 'valid', 'test'))
+            By default, all the three datasets (train, test, valid) are generated. Users
+            could also choose any one or two of them, for example ('train', 'test') or
+            just a string 'train'. If 'train' is not in the tuple or string, a vocab
+            object should be provided which will be used to process valid and/or test
+            data.
+        root: Directory where the datasets are saved. Default: ".data"
+        vocabs: A list of voabularies for each columns in the dataset. Must be in an
+            instance of List
+            Default: None
+        word_tokenizer: The tokenizer used to preprocess word column in raw text data
+            Default: None
```

Comment:
To keep consistent, we have `root`, `vocab`, `tokenizer`, `data_select` in order.